File: problems/HackerRank/2521_exercises/linkedList/easy/removeElements.c
struct ListNode *removeElements(struct ListNode *head, int val)
{

	struct ListNode *dummyNode = (struct ListNode *)malloc(sizeof(struct ListNode));
	dummyNode->next = head;
	struct ListNode *current = dummyNode;

	while (current->next != NULL)
	{
		if (current->next->val == val)
		{
			current->next = current->next->next;
		}
		else
		{
			current = current->next;
		}
	}
	return dummyNode->next;
}

File: problems/LeetCode/linked_list/removeElements.py
def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:

		dummyNode = ListNode(next = head)
		cur = dummyNode

		while cur.next:

				if cur.next.val == val:
						cur.next = cur.next.next
				else:
						cur = cur.next
		
		return dummyNode.next